.github/workflows/benchmark.yaml

```diff
@@ -0,0 +1,72 @@
+name: Benchmarks
+
+on:
+  pull_request:
+    branches: [ '**' ]
+    types: [opened, synchronize, reopened, labeled]
+
+env:
+  PROMSCALE_LOGGING: false
+  PROMSCALE_BENCHMARK: true
+  golang-version: 1.18.4
+
+jobs:
+  benchmark:
+    if: contains( github.event.pull_request.labels.*.name, 'action:benchmarks')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout current branch
+        uses: actions/checkout@v3
+
+      - name: Set up Go ${{ env.golang-version }}
+        uses: actions/setup-go@v3.2.1
+        with:
+          go-version: ${{ env.golang-version }}
+        id: go
+
+      - name: Benchmark current PR
+        run: go test -bench=. -run=^$ -benchmem -cpu=1 ./... | tee current.txt
```

Comment:
Continuing the [discussion from the previous PR](https://github.com/timescale/promscale/pull/1510/files#r926485351). I've had a bit more opportunity to fiddle around with this, and I think that we should increase the number of runs that we do (with `-count=X`), and we should not use `-delta-test=none` in `benchstat`. The delta test gives an indication of the statistical significance of the result, which is important in evaluating whether the numbers that we're looking at mean something or not.